File: src/main/java/com/github/cao/awa/lunaria/pool/LunariaPool.java
package com.github.cao.awa.lunaria.pool;

import java.util.concurrent.ForkJoinPool;
import java.util.concurrent.TimeUnit;

public class LunariaPool {
    private static final ForkJoinPool POOL = new ForkJoinPool();

    public static boolean awaitQuiescence(long timeout, TimeUnit unit) {
       return POOL.awaitQuiescence(timeout, unit);
    }

    public static boolean awaitTermination(long timeout, TimeUnit unit) throws InterruptedException {
        return POOL.awaitTermination(timeout, unit);
    }
}

File: src/main/kotlin/com/github/cao/awa/lunaria/pool/LunariaPool.kt
package com.github.cao.awa.lunaria.pool

import java.util.concurrent.ForkJoinPool
import java.util.concurrent.TimeUnit

object LunariaPool {
    private val pool: ForkJoinPool = ForkJoinPool()

    fun awaitQuiescence(timeout: Long, unit: TimeUnit) {
        this.pool.awaitQuiescence(timeout, unit)
    }

    fun awaitTermination(timeout: Long, unit: TimeUnit) {
        this.pool.awaitTermination(timeout, unit)
    }
}